### PR TITLE
fix: moved tsdx from dependencies to devDependencies

### PR DESCRIPTION
**Pull request by pgray-hiro**
## Description

This PR moves the `tsdx` package from `dependencies` to `devDependencies`.  This should resolve an error with the Todos App dev tutorial where a downstream dependency was failing a yarn preflight check.

For more details, see https://github.com/blockstack/docs/issues/1170.

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] API reference/documentation update
- [ ] Other

## Does this introduce a breaking change?
No

## Are documentation updates required?
No

## Checklist
- [x] Code is commented where needed
- [x] Unit test coverage for new or modified code paths
- [x] `yarn lerna run test` passes
- [x] Changelog is updated
- [x] Tag 1 of @hstove or @kyranjamie or @aulneau for review

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -65,8 +66,7 @@
     "is-regex": "^1.1.2",
     "jsx-ast-utils": "^3.2.0",
     "patch-package": "^6.4.7",
-    "postinstall-postinstall": "^2.1.0",
-    "tsdx": "^0.14.1"
+    "postinstall-postinstall": "^2.1.0"
```
Comment (aulneau) [inline package.json:69]: while we're at it, the dependencies should only be:
```    "bn.js": "^5.2.0",    "buffer": "^6.0.3",    "is-regex": "^1.1.2",    "jsx-ast-utils": "^3.2.0",```

**Review by aulneau — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (aulneau) [reply package.json:69]: everything else _should_ be a dev dependency if i'm not mistaken

**Review by fbwoolf — COMMENTED** [review]
`tsdx` is also in the dependencies in the connect package itself.

**Comment by pgray-hiro** [comment]
@aulneau Do I need to make a changeset here? The bot is nagging me. I'll update the dependencies.

**Comment by pgray-hiro** [comment]
Hopefully I did that right, my terminal output said I only did a minor bump to `@stacks-connect` but the bot is saying there's a major bump to `@stacks/connect-react` - not sure why.

File: .changeset/ninety-teachers-hunt.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@stacks/connect': minor
+---
+
+updated dependencies
```
Comment (aulneau) [inline .changeset/ninety-teachers-hunt.md:5]: Let's try to keep this a bit more verbose if possible, https://github.com/blockstack/connect/blob/main/packages/connect/CHANGELOG.md we want to make sure reading the changelog is fun and enjoyable!

maybe something like this:

> This update moves `tsdx` and other dependencies to the correct category of `devDependencies`, fixing a downstream issue in the todos example app.

Comment (aulneau) [reply .changeset/ninety-teachers-hunt.md:5]: You can edit this markdown file after the initial `yarn changeset` command to be whatever you like.

**Comment by pgray-hiro** [comment]
Sounds good, updated.

**Pull request merged**